cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelperTest.java

```diff
@@ -538,18 +541,26 @@ private void expectIncrementalBrokerConfigs(AdminClient adminClient, List<Intege
     }
   }
 
-  private void assertExpectedThrottledRateForBroker(int brokerId, Long expectedRate) throws ExecutionException, InterruptedException {
+  private void assertExpectedThrottledRateForBroker(int brokerId, Long expectedRate) {
     ConfigResource cf = new ConfigResource(ConfigResource.Type.BROKER, String.valueOf(brokerId));
-    Map<ConfigResource, Config> brokerConfig = _adminClient.describeConfigs(Collections.singletonList(cf)).all().get();
-    String expectedString = expectedRate == null ? null : String.valueOf(expectedRate);
-    assertNotNull(brokerConfig.get(cf));
-    if (expectedRate == null) {
-      assertNull(brokerConfig.get(cf).get(ReplicationThrottleHelper.LEADER_THROTTLED_RATE));
-      assertNull(brokerConfig.get(cf).get(ReplicationThrottleHelper.FOLLOWER_THROTTLED_RATE));
-    } else {
-      assertEquals(expectedString, brokerConfig.get(cf).get(ReplicationThrottleHelper.LEADER_THROTTLED_RATE).value());
-      assertEquals(expectedString, brokerConfig.get(cf).get(ReplicationThrottleHelper.FOLLOWER_THROTTLED_RATE).value());
-    }
+    waitUntilTrue(() -> {
```

Comment:
We are adding this retry logic for the unit tests, which ensures that the eventual response will settle and provide the expected response. But IIUC, the `ReplicationThrottleHelper#getEntityConfigs` can still return an unexpected result, causing failure for the main code.

Am I missing something?

Reply:
@efeg You're right, we ought do this in main instead of tests.

I initially thought it did not matter if we couldn't immediately read our writes but there are a number of scenarios where this is problematic. Also this keeps the existing semantics that were provided by ZooKeeper. So I agree with your feedback and I've added logic to ensure config changes are fully applied before we continue.